ksql-streams/src/main/java/io/confluent/ksql/execution/streams/ExecutionStepFactory.java

```diff
@@ -78,4 +109,275 @@ private ExecutionStepFactory() {
+      final ExecutionStep<KTable<K, GenericRow>> right,
+      final LogicalSchema resultSchema
+  ) {
+    final QueryContext queryContext = stacker.getQueryContext();
+    return new TableTableJoin<>(
+        new DefaultExecutionStepProperties(queryContext.toString(), resultSchema, queryContext),
+        joinType,
+        left,
+        right
+    );
+  }
+
+  public static <K> StreamAggregate<KTable<K, GenericRow>, KGroupedStream<Struct, GenericRow>>
+      streamAggregate(
+          final QueryContext.Stacker stacker,
+          final ExecutionStep<KGroupedStream<Struct, GenericRow>> sourceStep,
+          final LogicalSchema resultSchema,
+          final Formats formats,
+          final int nonFuncColumnCount,
+          final List<FunctionCall> aggregations
+  ) {
+    final QueryContext queryContext = stacker.getQueryContext();
+    return new StreamAggregate<>(
+        new DefaultExecutionStepProperties(queryContext.toString(), resultSchema, queryContext),
+        sourceStep,
+        formats,
+        nonFuncColumnCount,
+        aggregations
+    );
+  }
+
+  public static <K> StreamGroupBy<KStream<K, GenericRow>, KGroupedStream<Struct, GenericRow>>
+      streamGroupBy(
+          final QueryContext.Stacker stacker,
+          final ExecutionStep<KStream<K, GenericRow>> sourceStep,
+          final Formats format,
+          final List<Expression> groupingExpressions
+  ) {
+    final QueryContext queryContext = stacker.getQueryContext();
+    return new StreamGroupBy<>(
+        new DefaultExecutionStepProperties(
+            queryContext.toString(),
+            sourceStep.getProperties().getSchema(),
+            queryContext
+        ),
+        sourceStep,
+        format,
+        groupingExpressions
+    );
+  }
+
+  public static TableAggregate<KTable<Struct, GenericRow>, KGroupedTable<Struct, GenericRow>>
+      tableAggregate(
+          final QueryContext.Stacker stacker,
+          final ExecutionStep<KGroupedTable<Struct, GenericRow>> sourceStep,
+          final LogicalSchema resultSchema,
+          final Formats formats,
+          final int nonFuncColumnCount,
+          final List<FunctionCall> aggregations
+  ) {
+    final QueryContext queryContext = stacker.getQueryContext();
+    return new TableAggregate<>(
+        new DefaultExecutionStepProperties(queryContext.toString(), resultSchema, queryContext),
+        sourceStep,
+        formats,
+        nonFuncColumnCount,
+        aggregations
+    );
+  }
+
+  public static <K> TableGroupBy<KTable<K, GenericRow>, KGroupedTable<Struct, GenericRow>>
+      tableGroupBy(
+          final QueryContext.Stacker stacker,
+          final ExecutionStep<KTable<K, GenericRow>> sourceStep,
+          final Formats format,
+          final List<Expression> groupingExpressions
+  ) {
+    final QueryContext queryContext = stacker.getQueryContext();
+    return new TableGroupBy<>(
+        new DefaultExecutionStepProperties(
```

Comment:
I've seen this pattern a lot. It will get more verbose as more fields are added to `DefaultExecutionStepProperties` - consider having a
```suggestion
        sourceStep.getProperties().withQueryContext(queryContext);
```